src/palace/manager/integration/license/opds/opds1/extractor.py

```diff
@@ -0,0 +1,753 @@
+from __future__ import annotations
+
+from collections.abc import Generator, Sequence
+from dataclasses import dataclass
+from datetime import date, datetime
+from typing import Any
+from urllib.parse import urljoin
+
+import dateutil
+import feedparser
+from lxml import etree
+from lxml.etree import ElementTree, _Element as Element
+
+from palace.manager.core.exceptions import PalaceValueError
+from palace.manager.data_layer.bibliographic import BibliographicData
+from palace.manager.data_layer.circulation import CirculationData
+from palace.manager.data_layer.contributor import ContributorData
+from palace.manager.data_layer.identifier import IdentifierData
+from palace.manager.data_layer.link import LinkData
+from palace.manager.data_layer.measurement import MeasurementData
+from palace.manager.data_layer.subject import SubjectData
+from palace.manager.integration.license.opds.bearer_token_drm import BearerTokenDrmMixin
+from palace.manager.integration.license.opds.data import FailedPublication
+from palace.manager.integration.license.opds.extractor import (
+    OpdsExtractor,
+)
+from palace.manager.integration.license.opds.opds1.settings import (
+    IdentifierSource,
+)
+from palace.manager.integration.license.opds.opds1.xml_parser import OPDSXMLParser
+from palace.manager.opds.odl.info import LicenseInfo
+from palace.manager.opds.odl.odl import License
+from palace.manager.sqlalchemy.model.classification import Subject
+from palace.manager.sqlalchemy.model.edition import Edition
+from palace.manager.sqlalchemy.model.licensing import LicensePool, RightsStatus
+from palace.manager.sqlalchemy.model.measurement import Measurement
+from palace.manager.sqlalchemy.model.resource import Hyperlink
+from palace.manager.util import first_or_default
+from palace.manager.util.datetime_helpers import datetime_utc, utc_now
+from palace.manager.util.opds_writer import OPDSFeed
+
+
+@dataclass
+class OPDS1Feed:
+    parser: OPDSXMLParser
+    feed_parser: dict[str, Any]
+    etree: ElementTree
+
+
+@dataclass
+class OPDS1Publication:
+    parser: OPDSXMLParser
+    entry_fp: dict[str, Any]
+    entry_xml: Element
+
+
+class Opds1Extractor(OpdsExtractor[OPDS1Feed, OPDS1Publication], BearerTokenDrmMixin):
```

Comment:
The parsing logic in this class mostly comes from the old `OPDSImporter` class. 

I did a major refactor when I was turning this into an extractor class to break up the `extract_data_from_feedparser` and `extract_bibliographic_from_elementtree` functions into finer grained functions that extract a single piece of data while keeping the parsing logic from those functions.